### PR TITLE
Ft new profile

### PR DESCRIPTION
**Pull request by Abigail-cloud**
I was able to commit the changes pointed out ... Removed dead code and installed tslog package

### CODE REVIEW

File: src/Repository/Auth/index.ts [hunk 1]
```diff
@@ -1,9 +1,64 @@
...
+
+
+
+
+
+
+
+
+
+  // public async hashedPassword (password:string): Promise<any>{
```
Comment (forestbaba) [inline src/Repository/Auth/index.ts]: Remove dead codes

File: src/middleware/auth.ts [hunk 2]
```diff
@@ -20,11 +22,14 @@ export const verifyToken = (
...
         return res.status(403).send("A token is required for authentication");
     }
-    try {
         const decoded = jwt.verify(token, TOKEN_SECRET);
-        console.log(decoded);
+        
+    req.body.authUser = decoded
+    
     } catch (err) {
+        console.log(err);
```
Comment (forestbaba) [inline src/middleware/auth.ts]: Replace console.log with logger

File: src/models/userModel.ts [hunk 3]
```diff
@@ -1,18 +1,23 @@
...
     {
         firstName: { type: String, required: true },
         lastName: { type: String, required: true },
         email: { type: String, required: true },
         mobile: { type: Number, required: true },
-        password: { type: String, required: true },
+        password: { type: String, required: true},
         active: { type: Boolean, required: true },
-        verificationCode: { type: String, required: true, unique: true },
+        verificationCode: { type: String, required: true, unique: true, select:false },
```
Comment (forestbaba) [inline src/models/userModel.ts]: This should not be unique and should not be required

File: src/server.ts [hunk 4]
```diff
@@ -1,5 +1,10 @@
 import express, { Application, Request, Response } from "express";
 import cors from "cors";
+import * as dotenv from 'dotenv'
+import database from "./config/database";
+import { authRoute } from "./routes/authRoutes";
+import userRouter from './routes/userProfile';
+=======
```
Comment (forestbaba) [inline src/server.ts]: What's this line doing here?...Are you sure this is not git conflict?

File: src/server.ts [hunk 5]
```diff
@@ -40,6 +45,8 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
 authRoute(app);
+app.use("/",userRouter)
+=======
```
Comment (forestbaba) [inline src/server.ts]: Same here

File: src/services/Auth/index.ts [hunk 6]
```diff
@@ -16,69 +16,63 @@ const auth = new AuthRepository();
...
         email: req.body.email,
         mobile: req.body.mobile,
         password: bcrypt.hashSync(req.body.password, 8),
         active: false,
         verificationCode: code,
-    });
-
-    const isUserExist = await User.findOne({ email: newUser.email });
+    };
+    const isUserExist = await auth.checkUserExist (req.body.email);
```
Comment (forestbaba) [inline src/services/Auth/index.ts]: change auth to authRepository so it depicts the name

File: src/services/Auth/index.ts [hunk 7]
```diff
@@ -16,69 +16,63 @@ const auth = new AuthRepository();
...
 
+
 export const loginService = async (req: Request, res: Response) => {
-    const user = await User.findOne({ email: req.body.email });
+    
+    const user = await auth.loginUser(req.body.email);
+    
     if (!user) return res.status(404).json({ message: "User Not found." });
 
     const vp = bcrypt.compareSync(req.body.password, user.password);
```
Comment (forestbaba) [inline src/services/Auth/index.ts]: What's vp??...Please give it a descriptive name

File: src/services/profile/viewUser.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { Request, Response } from "express";
+import { AuthRepository } from "../../Repository/Auth";
+const auth = new AuthRepository();
+
+const viewCurrentUser = async (req:Request, _res:Response) => {
+    const getuser = auth.getUser(req.params.id)
```
Comment (forestbaba) [inline src/services/profile/viewUser.ts]: Change this name as recommended above

File: src/Repository/Auth/index.ts (hunk 1 above)
Comment (Abigail-cloud) [reply src/Repository/Auth/index.ts]: Dead codes were deleted

File: src/middleware/auth.ts (hunk 2 above)
Comment (Abigail-cloud) [reply src/middleware/auth.ts]: installed tslog package

File: src/models/userModel.ts (hunk 3 above)
Comment (Abigail-cloud) [reply src/models/userModel.ts]: Removed both required and unique field

File: src/server.ts (hunk 4 above)
Comment (Abigail-cloud) [reply src/server.ts]: Resolvedconflicts

File: src/server.ts (hunk 5 above)
Comment (Abigail-cloud) [reply src/server.ts]: removed conflicts

File: src/services/Auth/index.ts (hunk 6 above)
Comment (Abigail-cloud) [reply src/services/Auth/index.ts]: Changed it to authRepository class

File: src/services/Auth/index.ts (hunk 7 above)
Comment (Abigail-cloud) [reply src/services/Auth/index.ts]: Changed it to hashedPassword

File: src/services/profile/viewUser.ts (hunk 8 above)
Comment (Abigail-cloud) [reply src/services/profile/viewUser.ts]: changed auth to new AuthRepository () as it's constructor signature

**Review by forestbaba — APPROVED** [review]

**Pull request merged**